internal/typeinfo/validate.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under Apache 2.0, see LICENCE file for details.
+
+package typeinfo
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type TypeToValue map[reflect.Type]reflect.Value
```

Comment:
This is a type definition, rather than a type alias.
An alias would be `type TypeToValue = map[reflect.Type]reflect.Value` (with an equals sign).

I tend to prefer using an alias if you don't need to further extend the type because:
- You can assign one to the other directly without casting.
- You can compare them.

Reply:
Ah yea, I'm so used to only using type definitions because SQLair doesn't work with aliases! Changed.